app/src/test/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileRepoTest.kt

```diff
@@ -0,0 +1,112 @@
+class PinnedTileRepoTest {
+
+    private lateinit var pinnedTileRepo: PinnedTileRepo
+
+    @Before
+    fun setUp() {
+        pinnedTileRepo = PinnedTileRepo(RuntimeEnvironment.application)
+    }
+
+    @Test
+    fun `WHEN repo fetches initial tileList THEN repo should emit a feed of only bundled tile list`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assertEquals(10, (it as LinkedHashMap<String, PinnedTile>).size)
+        })
+
+        pinnedTileRepo.getPinnedTiles().observeForever(observerSpy)
+        pinnedTileRepo.loadTilesCache()
+
+        verify(observerSpy, times(1)).onChanged(any())
+    }
+
+    @Test
+    fun `WHEN repo adds a new pin THEN repo should emit a combined list of bundled and custom`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assert((it as LinkedHashMap<String, PinnedTile>).size == 10 || it.size == 11)
+        })
+
+        pinnedTileRepo.getPinnedTiles().observeForever(observerSpy)
+        pinnedTileRepo.loadTilesCache()
+        assertEquals(0, pinnedTileRepo.customTilesSize)
+        pinnedTileRepo.addPinnedTile("https://example.com", null)
+        assertEquals(1, pinnedTileRepo.customTilesSize)
+
+        verify(observerSpy, times(2)).onChanged(any())
+    }
+
+    @Test
+    fun `WHEN repo removes a bundled pin THEN repo should emit a smaller list and bundled size`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assert((it as LinkedHashMap<String, PinnedTile>).size == 10 || it.size == 9)
+        })
+
+        pinnedTileRepo.getPinnedTiles().observeForever(observerSpy)
+        pinnedTileRepo.loadTilesCache()
+        assertEquals(10, pinnedTileRepo.bundledTilesSize)
+        pinnedTileRepo.removePinnedTile("https://pinterest.com/")
+        assertEquals(9, pinnedTileRepo.bundledTilesSize)
+
+        verify(observerSpy, times(2)).onChanged(any())
+    }
+
+    @Test
+    fun `WHEN repo fails to remove a bundled pin THEN repo should maintain same bundled size`() {
+        val observerSpy = spy(Observer<LinkedHashMap<String, PinnedTile>> {
+            assertTrue(it is LinkedHashMap<String, PinnedTile>)
+            assert((it as LinkedHashMap<String, PinnedTile>).size == 10)
+        })
+
+        pinnedTileRepo.getPinnedTiles().observeForever(observerSpy)
+        pinnedTileRepo.loadTilesCache()
+        assertEquals(10, pinnedTileRepo.bundledTilesSize)
+        pinnedTileRepo.removePinnedTile("https://example.com/")
+        assertEquals(10, pinnedTileRepo.bundledTilesSize)
+
+        verify(observerSpy, times(1)).onChanged(any())
+    }
+
+    @Test
+    fun `WHEN repo removes a custom pin THEN repo should emit a smaller list and custom size`() {
```

Comment:
I don't think this should block the merge, but in the future we could decouple these tests from the particulars of our bundled tiles by breaking out a separate object for retrieving them from the file system and injecting it through the repos constructor.  That way the test could control how many exist, and we could also give them useful mock data.